### DISCUSSION TITLE
Convert a storybook meta title to a valid storybook url

### DISCUSSION THREAD

**Discussion by selenecodes** [storybookjs/storybook#25163] [Help]
### Summary

We use storybook to document our components but we also have a lot of typescript code which needs to be documented. To prevent our docs from getting scattered we currently use typedoc with a custom plugin that converts each TypeDoc markdown file into a storybook.mdx file. 

**Question:**
Storybook somehow converts a storybook meta title to a URL for linking between stories, I've mostly got this working but was wondering if there was a function I could use to automatically convert a storybook meta title to a valid URL. (without writing my own custom regex which might get out of sync with storybook over time*)

### Additional information

An example of what's currently happening:
1. I create a file
2. I set the Meta title to be `Storybook folder prefix` + `File basename`
3. Storybook somehow converts this meta title to a valid storybook url
4. My own code to replace markdown urls does this incorrectly.
5. E.g. does anyone know what the function is that storybook uses to convert meta title to a url

| Filename                   | Meta Title                     | Storybook URL                                   | My url currently                                 |
|----------------------------|--------------------------------|-------------------------------------------------|--------------------------------------------------|
| routes__layout.stories.mdx | TypeDoc/modules/routes__layout | ?path=/docs/typedoc-modules-routes-layout--docs | ?path=/docs/TypeDoc-modules-routes__layout--docs |
| reset.stories.mdx          | TypeDoc/modules/reset          | ?path=/docs/typedoc-modules-reset--docs         | ?path=/docs/TypeDoc-modules-reset--docs          |
| lib_example.stories.mdx    | TypeDoc/modules/lib_example    | ?path=/docs/typedoc-modules-lib-example--docs   | ?path=/docs/TypeDoc-modules-lib_example--docs    |

**Comment by chantastic** [comment]
https://github.com/ComponentDriven/csf may be a helpful place to start.
